src/main/java/com/commercetools/project/sync/Syncer.java

```diff
@@ -28,39 +35,41 @@
- *     etc..)
- * @param <RC> The type of the resource to create (e.g. {@link io.sphere.sdk.products.Product},
- *     {@link io.sphere.sdk.categories.Category}, etc..)
- * @param <RD> The type of the resource draft (e.g. {@link io.sphere.sdk.products.ProductDraft},
- *     {@link io.sphere.sdk.categories.CategoryDraft}, etc..)
- * @param <S> The type of the sync statistics resulting from the sync process (e.g. {@link
- *     com.commercetools.sync.products.helpers.ProductSyncStatistics}, {@link
+ * @param <ResourceT> The type of the resource to update (e.g. {@link ProductProjection}, {@link
+ *     Category}, etc..)
+ * @param <ResourceUpdateActionT> The update actions to update the resource with (e.g.
+ * {@link com.commercetools.api.models.product.ProductUpdateAction})
+ * @param <ResourceDraftT> The type of the resource draft (e.g. {@link ProductDraft}, {@link
+ *     CategoryDraft}, etc..)
+ * @param <SyncStatisticsT> The type of the sync statistics resulting from the sync process (e.g.
+ *     {@link com.commercetools.sync.products.helpers.ProductSyncStatistics}, {@link
  *     com.commercetools.sync.categories.helpers.CategorySyncStatistics}, etc..)
- * @param <O> The type of the sync options used for the sync (e.g. {@link
+ * @param <SyncOptionsT> The type of the sync options used for the sync (e.g. {@link
  *     com.commercetools.sync.products.ProductSyncOptions}, {@link
  *     com.commercetools.sync.categories.CategorySyncOptions}, etc..)
- * @param <Q> The type of the query used to query for the source resources (e.g. {@link
- *     io.sphere.sdk.products.queries.ProductProjectionQuery}, {@link
- *     io.sphere.sdk.categories.queries.CategoryQuery}, etc..)
- * @param <B> The type of the sync instance used to execute the sync process (e.g. {@link
+ * @param <PagedQueryT> The type of the query used to query for the source resources (e.g. {@link
+ *     com.commercetools.api.client.ByProjectKeyProductProjectionsGet}, {@link
+ *     com.commercetools.api.client.ByProjectKeyCategoriesGet}, etc..)
+ * @param <BaseSyncT> The type of the sync instance used to execute the sync process (e.g. {@link
  *     com.commercetools.sync.products.ProductSync}, {@link
  *     com.commercetools.sync.categories.CategorySync}, etc..)
  */
 public abstract class Syncer<
-    RU extends ResourceView,
-    RC extends ResourceView<RC, RC>,
-    RD,
-    V extends Versioned,
-    S extends BaseSyncStatistics,
-    O extends BaseSyncOptions<RU, RD, RC>,
-    Q extends QueryDsl<RU, Q>,
-    B extends BaseSync<RD, V, S, O>> {
+    ResourceT extends BaseResource & DomainResource<ResourceT> & WithKey,
+    ResourceUpdateActionT extends ResourceUpdateAction<ResourceUpdateActionT>,
+    ResourceDraftT,
+    SyncStatisticsT extends BaseSyncStatistics,
+    SyncOptionsT extends BaseSyncOptions<ResourceT, ResourceDraftT, ResourceUpdateActionT>,
+    PagedQueryT extends PagedQueryResourceRequest<PagedQueryT, PagedQueryResponseT>,
```

Comment:
Please use non-deprecated version of `PagedQueryResourceRequest`.

Reply:
I've already tried to use the non-deprecated but it's conflicting with the QueryAll utils in sdk. I will create a ticket to have a look at this later.